ox_bqpipeline/bqpipeline.py

```diff
@@ -297,9 +356,66 @@ def create_job_config(self, batch=False, dest=None, create=True,
                                        create_disposition=create_disp,
                                        write_disposition=write_disp)
 
+    def validate_query_params(self, query_params, depth=0):
+        """Validate the named/positional query parameters."""
+        if not (isinstance(query_params, list) or
+                isinstance(query_params, dict)):
+            return False
+        for key in query_params:
+            if isinstance(query_params, list):
+                value = key
+                key = None
+            else:
+                if not (depth or isinstance(key, str)):
+                    return False
+                value = query_params[key]
+
+            # For list, make sure each element is a scalar type.
+            # For dict, check recursively for valid parameters.
+            # for others, check that they map to valid scalar types.
+            if isinstance(value, list):
+                typ = set([type(val).__name__ for val in value])
+                res = not typ or (len(typ) == 1 and
+                                  typ.pop() in bq_scalar_type_map)
+            elif isinstance(value, dict):
+                res = self.validate_query_params(value, depth=1)
+            else:
+                res = type(value).__name__ in bq_scalar_type_map
```

Comment:
don't use `__name__` (see suggested refactoring of bq_scalar_type_map)

Reply:
Done.